attendance_app/models.py

```diff
@@ -9,14 +59,65 @@ class UserProfile(models.Model):
     email = models.TextField()
     created_on = models.DateTimeField(auto_now_add = True)
     role = models.CharField(max_length = 40)
+    objects = UserProfileManager()
+
+    def __str__(self):
+        return self.username
+
+class AttendanceManager(models.Manager):
+    def createAttendance(self, created_by, created_on):
+        attendance=self.create(created_by=created_by,created_on=created_on)
+        return attendance.id
+
+    def getAttendanceByID(self, attendanceID):
+        try:
+            attendance = Attendance.objects.get(id__exact=attendanceID)
+            return attendance
+        except MultipleObjectsReturned:
+            print ("More than one objects with the same username and chat_url.")
+            return None
+        except DoesNotExist:
+            print ("Object does not exist.")
+            return None   
+
 
 class Attendance(models.Model):
-	created_by = models.ForeignKey(UserProfile, on_delete = models.SET_NULL, null = True)
-	created_on = models.DateTimeField(auto_now_add = True)
+    created_by = models.ForeignKey(UserProfile, on_delete = models.SET_NULL, null = True)
+    created_on = models.DateTimeField(auto_now_add = True)
+    objects = AttendanceManager()    
+
+class AttendanceSubmitManager(models.Model):
+    def createAttendanceSubmit(self, attendance, submitted_on, username, chat_url,
+                                first_name, last_name, email, created_on, role):
 
+        submitted_by = UserProfile.objects.createUserProfile(username=username, chat_url=chat_url, 
```

Comment:
createAttendanceSubmit should not need submitted_on and created_on
Use Builder design pattern